Add function `primary_group()` to `modules/pw_user.py` and `get_group_name()` to `utils/user.py`

gluf-technik opened the pull request:
### What does this PR do?
Adds the missing function `primary_group(name)` to `modules/pw_user.py` so one can use `salt['user.primary_group'](username)` on FreeBSD, too.
As functions from `salt.utils.user` are used anyway I've also added a function to get the name to a given group ID there, `salt.utils.user.get_group_name(gid)` (as I misunderstood `utils.user.get_default_group()` and thought it would return a group id instead of the primary/default group's name).

### What issues does this PR fix or reference?
Fixes: None

### Previous Behavior
```
kvm-master:~# salt -C "*" user.primary_group root
deb11-kvm-minion:
    root
alpine317-kvm-minion:
    root
[...]
obsd72-kvm-minion:
    wheel
fbsd13-kvm-minion:
    'user.primary_group' is not available.
```
### New Behavior
```
kvm-master:~# salt -C "G@os:FreeBSD" user.primary_group root
fbsd13-kvm-minion:
    wheel
```

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**
<!-- Please review the [test documentation](https://docs.saltproject.io/en/master/topics/tutorials/writing_tests.html) for details on how to implement tests into Salt's test suite. -->
- [ ] Docs
- [x] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [ ] Tests written/updated

### Commits signed with GPG?
No

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

gluf-technik committed:
Add `primary_group()` to `modules/pw_user.py`

Just like [`primary_group(name)` in `modules/useradd.py`](https://docs.saltproject.io/en/latest/ref/modules/all/salt.modules.useradd.html#salt.modules.useradd.primary_group) but using functions from `salt.utils.user` instead of [`id(1)`](https://man.freebsd.org/cgi/man.cgi?id)

gluf-technik committed:
Add get_group_name(gid) to utils/user.py

Using Python's `grp.getgrgid()` to turn a group ID into the groups name.

gluf-technik committed:
Don't try to convert a group name to a group name

`salt.utils.default_group()` actually returns a group name, not a group id, so there's no reason to try to convert its return value from a GID to a group name.

gluf-technik committed:
Forgot to run pylint on `utils/user.py`

gluf-technik committed:
Changelog entry for PW #63899 "Add function primary_group() to modules/pw_user.py and get_group_name() to utils/user.py"

dwoz commented:
@gluf-technik Can you please add a simple test case for this?

twangboy commented on salt/utils/user.py:
```diff
@@ -339,6 +339,18 @@ def get_group_list(user, include_default=True):
     return sorted(ugroups)
 
 
+def get_group_name(gid):
```
This function doesn't appear to be called by anything

twangboy reviewed (changes requested):
This needs some tests for both the util and the module

dwoz committed:
Add tests for pw_user.primary_group and utils.user.get_group_name

- Two unit tests for pw_user.primary_group (happy path + nonexistent user)
- Two unit tests for utils.user.get_group_name (known gid + unknown gid)
- Fix docstring formatting in pw_user.primary_group (missing blank lines,
  add versionadded:: 3009.0 to match convention in mac_user/useradd)
- Fix changelog typos: py_user.py -> pw_user.py, utils/users.py -> utils/user.py
